docs/migration-v8.md

```diff
@@ -1 +1,515 @@
+    async cancellationToken =>
+    {
+        // your code here
+    },
+    cancellationToken);
+
+// Create and use the ISyncPolicy<T>
+ISyncPolicy<HttpResponseMessage> syncPolicyT = Policy
+    .HandleResult<HttpResponseMessage>(r => !r.IsSuccessStatusCode)
+    .WaitAndRetry(3, _ => TimeSpan.FromSeconds(1));
+
+syncPolicyT.Execute(() =>
+{
+    // your code here
+    return GetResponse();
+});
+
+// Create and use the IAsyncPolicy<T>
+IAsyncPolicy<HttpResponseMessage> asyncPolicyT = Policy
+    .HandleResult<HttpResponseMessage>(r => !r.IsSuccessStatusCode)
+    .WaitAndRetryAsync(3, _ => TimeSpan.FromSeconds(1));
+await asyncPolicyT.ExecuteAsync(
+    async cancellationToken =>
+    {
+        // your code here
+        return await GetResponseAsync(cancellationToken);
+    },
+    cancellationToken);
+```
+<!-- endSnippet -->
+
+Points to note from the example:
+
+- `Policy.Handle<Exception>()` determines which exceptions are handled.
+- `Policy.HandleResult<HttpResponseMessage>()` determines which results are managed.
+- The final `WaitAndRetry()` produces an `ISyncPolicy`, while `WaitAndRetryAsync()` results in an `IAsyncPolicy`.
+- The retry policy tries 3 times, waiting 1 second between attempts.
+
+### Configuring strategies in v8
+
+In Polly v8, the previous code becomes:
+
+<!-- snippet: migration-policies-v8 -->
+```cs
+// Create and use the ResiliencePipeline
+ResiliencePipeline pipeline = new ResiliencePipelineBuilder()
+    .AddRetry(new RetryStrategyOptions
+    {
+        ShouldHandle = new PredicateBuilder().Handle<Exception>(),
+        Delay = TimeSpan.FromSeconds(1),
+        MaxRetryAttempts = 3,
+        BackoffType = DelayBackoffType.Constant
+    })
+    .Build();
+
+// Synchronous execution
+pipeline.Execute(() =>
+{
+    // your code here
+});
+
+// Asynchronous execution
+await pipeline.ExecuteAsync(async cancellationToken =>
+{
+    // your code here
+},
+cancellationToken);
+
+// Create and use the ResiliencePipeline<T>
+ResiliencePipeline<HttpResponseMessage> pipelineT = new ResiliencePipelineBuilder<HttpResponseMessage>()
+    .AddRetry(new RetryStrategyOptions<HttpResponseMessage>
+    {
+        ShouldHandle = new PredicateBuilder<HttpResponseMessage>()
+            .Handle<Exception>()
+            .HandleResult(result => !result.IsSuccessStatusCode),
+        Delay = TimeSpan.FromSeconds(1),
+        MaxRetryAttempts = 3,
+        BackoffType = DelayBackoffType.Constant
+    })
+    .Build();
+
+// Synchronous execution
+pipelineT.Execute(() =>
+{
+    // your code here
+    return GetResponse();
+});
+
+// Asynchronous execution
+await pipelineT.ExecuteAsync(async cancellationToken =>
+{
+    // your code here
+    return await GetResponseAsync(cancellationToken);
+},
+cancellationToken);
+```
+<!-- endSnippet -->
+
+Here are the primary changes:
+
+- `ResiliencePipelineBuilder` is used to start building the resilience pipeline, instead of the static `Policy.HandleException<TException>(...)`.
```

Comment:
I think these things should be code comments as well, so that readers don't have to go back and forth to understand things. It's always easier to understand what has changed if you can see the differences in smaller chunks rather than at bulk.

Reply:
Done.